crates/matrix-sdk/src/http_client.rs

```diff
@@ -297,7 +297,10 @@ impl HttpSettings {
             let user_agent =
                 self.user_agent.clone().unwrap_or_else(|| "matrix-rust-sdk".to_owned());
 
-            http_client = http_client.user_agent(user_agent).timeout(self.timeout);
+            http_client = http_client
+                .user_agent(user_agent)
+                .timeout(self.timeout)
+                .http2_keep_alive_interval(Some(Duration::from_secs(30u64)));
```

Comment:
Do we want to make this configurable? How confident are we that this is the right™ value for everybody?

Reply:
two good questions. 

I would want to enable it for everyone so _some_ number is needed. My personal experience is that most connection timeout between 30s and 2min of idle time either way, so a very large number has probably no real effect in keeping it actually alive. BUT considering that almost all our code has either a sliding-sync or regular sync run in a loop, in practice this number probably doesn't actually matter as it won't be hit either way because we are long-polling.

So, I'd say: it probably doesn't matter, we just want it turned on.

But we can consider exposing the exact value (maybe that matters for appservices?)

Reply:
Yeah, let's make this configurable, who knows what some weirder projects might need.